app/controllers/apps_controller.rb

```diff
@@ -25,6 +25,10 @@ def show
   def destroy
     @app = App.find(params[:id])
 
+    @app.services.each do |service|
+      RemoveServiceJob.perform_later(server, @app.clean_name, service)
```

Comment:
Nope, thats why I pass in the actual name of the app, and not an instance of app. The unlink command in the RemoveServiceJob might fail, but it will gracefully fail, and then perform the removal of the service (The last part is independent of the actual app)